From c7314f8a76f5eebef07010c492011b1a8ae9a848 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 21 Jan 2021 14:14:54 +0200
Subject: [PATCH] [SW-30548] preboot: uart changed to slow return

Measuring UART timings in preboot shows UART takes significant time,
~2.2msec for handling a key press.
It is therefore required that UART handling in preboot will be of type
"slow return". The "slow return" in preboot assures that while the slow
modules run, other HW "pipes" related to other modules
(such as i2c for example) are not getting overflowed in the meanwhile.

Change-Id: Ic99d6fcb1e0cc1c760e9301c3014f847147e12b7
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/preboot/module_uart.c b/preboot/module_uart.c
index 340a140..be24344 100644
--- a/preboot/module_uart.c
+++ b/preboot/module_uart.c
@@ -16,7 +16,7 @@
 
 	preboot_puts("Keyboard press received\n");
 
-	return FOUND_FAST_RETURN;
+	return FOUND_SLOW_RETURN;
 }
 
 __weak void hl_esc_key_handle(void) {}
diff --git a/preboot/preboot.c b/preboot/preboot.c
index 0b118ab..d351223 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -62,9 +62,9 @@
 		.setup = noop,
 		.check = module_uart_check,
 		.disable = noop,
-		.run = noop,
+		.run = module_uart_run,
 		.enable = noop,
-		.run_fast = module_uart_run,
+		.run_fast = noop,
 		.teardown = noop,
 		.run_noreturn = noop,
 	},
